src/writer.rs

```diff
@@ -77,6 +62,84 @@ where
     Ok(())
 }
 
+/// Write a RGB nifti file (.nii or .nii.gz) in Little Endian.
+///
+/// If a `reference` is given, it will be used to fill most of the header's fields, except those
+/// necessary to be recognized as a RGB image. `scl_slope` will be set to 1.0 and `scl_inter` to
+/// 0.0.  If `reference` is not given, a default `NiftiHeader` will be built and written.
+pub fn write_rgb_nifti<P, D>(
+    path: P,
+    data: &Array<[u8; 3], D>,
+    reference: Option<&NiftiHeader>,
+) -> Result<()>
+where
+    P: AsRef<Path>,
+    D: Dimension + RemoveAxis,
+{
+    // Copy most of the reference header, but the `bitpix`, `scl_slope` and `scl_inter` fields must
```

Comment:
Good question. I choose those 3 because they were necessary. If I remove the UNUSED and all fields related to the affine, we have `intent` stuff, `dim_info`, slice_start/end/code, cal_min/max, `slice_duration`, `toffset` and `descrip`.

- `intent_code` and friends should be set to `0` I think. It would be `RGB: 2003` if we used the "rgb triplet in the 5th dimension", but we don't. I thought this was a hack but it's actually recognized by the nifti standard.
- `cal_min` and `cal_max` probably don't make much sense for a RGB image.
- `toffset` should always be set to 0.0 for 3D images, I guess. But it looks like an unimportant field.
- `descrip`, we just can't know.

The user can still give a reference header if he wants to modify some fields. So, do you want me to change something?

Reply:
Those don't pose an issue for the time being, I was mostly concerned about the three fields I mentioned above. Unless I overlooked it, these ones can still be overridden by the caller, and I can only picture this as room for disasters with no legitimate use cases.

`magic` should be set to the proper code if it's not already correct (`ni1\0` for headers, `n+1\0` for full volumes). `sizeof_hdr` should be always 348. And for now (since we don't support writing extensions), `vox_offset` should be always 352.